src/components/Table/components/cell.tsx

```diff
@@ -57,21 +57,16 @@ export const defaultRenderFn: (as: RenderAs) => RenderCellFn<any> =
   };
 
 /** Sets up the `GridContext` so that header cells can access the current sort settings. */
-export const headerRenderFn: (
-  columns: GridColumn<any>[],
-  column: GridColumn<any>,
-  sortState: SortState<any> | undefined,
-  setSortKey: Function | undefined,
-  as: RenderAs,
-) => RenderCellFn<any> =
-  (columns, column, sortState, setSortKey, as) =>
-  (key, css, content, row, rowStyle, classNames: string | undefined) => {
-    const [currentKey, direction] = sortState || [];
+export const headerRenderFn: (columns: GridColumn<any>[], column: GridColumn<any>, as: RenderAs) => RenderCellFn<any> =
+  (columns, column, as) => (key, css, content, row, rowStyle, classNames: string | undefined) => {
+    const { tableState } = useContext(RowStateContext);
+    const { current } = tableState.sortState ?? {};
```

Comment:
Question: Do we need to read this in a `useComputed` so that we re-render when it changes? Otherwise what drives the re-render of a header to flip up arrow to down arrow or what happen? 

...fwiw I think we could probably remove `GridSortContextProps` and have whoever was reading that read from `useContext(RowStateContext)` instead.

We don't have to remove `GridSortContextProps` in this PR, but I am curious about how this is re-rendering.

Reply:
Good question. I didn't think of this. It appears the whole header row is re-rendering. It says some hook in `RowImpl` is changing.. trying to figure out which.

Reply:
I can't seem to figure it out... It is really hard to track down hooks in `RowImpl` as all the hooks being triggered from the the cells, such as the `headerRenderFn` are considered a hook of the `RowImpl`. I have updated this to use a `useComputed`, but still going to try and track this down.